services/github/github-size.service.js

```diff
@@ -41,16 +60,25 @@ export default class GithubSize extends GithubAuthV3Service {
     }
   }
 
-  async fetch({ user, repo, path }) {
-    return this._requestJson({
-      url: `/repos/${user}/${repo}/contents/${path}`,
-      schema,
-      errorMessages: errorMessagesFor('repo or file not found'),
-    })
+  async fetch({ user, repo, path, branch }) {
+    if (branch) {
+      return this._requestJson({
+        url: `/repos/${user}/${repo}/contents/${path}?ref=${branch}`,
+        schema,
+        errorMessages: errorMessagesFor('repo, branch or file not found'),
+      })
+    } else {
+      return this._requestJson({
+        url: `/repos/${user}/${repo}/contents/${path}`,
+        schema,
+        errorMessages: errorMessagesFor('repo or file not found'),
+      })
+    }
```

Comment:
This is fine, but one thing to be aware of for future reference is that we typically like to use the `options` key on the arg which in turn allows our http client library to handle the conditionality of the query search parameters for us. I gather there's also a need to have slightly different error texts depending on the branch, though that could also be handled with a single call too, e.g.:

```js
    return this._requestJson({
      url: `/repos/${user}/${repo}/contents/${path}`,
      schema,
      options: { searchParams: { ref: branch } },
      errorMessages: errorMessagesFor(
        `repo${branch ? ', branch' : ''} or file not found`
      ),
    })
```

Reply:
That’s neat, I will try to remember about this syntax - thanks ☺️